### PR TITLE
Compact dashboard

### LINKED ISSUE #57: [Feature]: Remove extra space on Op Comp screen with place
### The feature request

Submitted by Ron KK1L via the Node Red Groups IO

image

### Proposed solution

Removing the group name?  

### Pages Impacted

_No response_

### Additional context

_No response_

### Submission

- [X] I agree this is the first post associated with this feature.

### LINKED ISSUE #59: [Bug]: Station Name switch for SO1V category in "UDP Radio Data XML" block does not use the correct variable
### Contact Details

kk1l@comcast.net

### What happened?

When creating a compact version of the Dashboard I noticed that the Station Name switch is using the N1MM+  logging station name to compare to the global variables PC1 & PC2 which are names chosen for the "Radios" on the Config tab.

In my pending (not yet requested) "compact-dashboard" pull request I added "msg.RadioNames=global.get('PCs');" to the UTF8 function to allow me to use the names in a label. Maybe a variation on this could be used to pipe in the correct variable.

image

### Version

1.2.20

### Environment

```markdown
- **OS** : 
- **Node.js** :
```

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by KK1L**
As a first step to compacting some of the screens I started with the main Dashboard. I integrated the radio titles into the Op label on the Radio1 and 2 info along the bottom and removed the titles from a few of the other blocks.

To allow the radio names to be displayed I added "msg.RadioNames=global.get('PCs');" to the UTF8 function to allow me to use the names the labels for Radio 1 Operator Dashboard and Radio 2 Operator Dashboard.

### CODE REVIEW

**Comment by KK1L** [comment]
Crap...i requested to pull into main not dev! darn...twice same mistake.

**Comment by KK1L** [comment]
image

**Comment by KK1L** [comment]
Regarding the merge of #59...the change I made to add labels associated with the radios is a global variable copy. Neither of those indicate which radio the data is for. Perhaps using "payload.RadioInfo.RadioNr[0]" like in switch "Radio 1 or 2" would work?? This would be for SO1V where only one radio is active. I am not sure the point of Radio 1 and Radio 2 in an SO1V environment.

**Comment by kd9lsv** [comment]
Ron, 

Your commits were not including the latest changes to dev. I rebased the branch upon dev to include the additional changes that I have made in between the other Tempateure PR and this requiring  forced pushing the changes up. and associated changes. 

Please pull these changes down before you create another commit. See below for git tree to clarify branches. 

image

I am still looking at the rest of the changes but for sure I know that 5 digit distance sizes are too big. 
image

I am also letting @kylekrieg  initially look at this for QA on the changes. 

Let me know if you have any issues.

**Comment by KK1L** [comment]
Thanks. I will get the hang of this. I was not offered any pushes when I committed. Probably missed a step. In see 9 pull and 9 pushes which I am merging on my local now.

**Comment by kd9lsv** [comment]
I still have a issue with what you did last time. You merge both commits which duplicated the same commits in series.  To become even with github, please force-pull (pull --force) using the terminal on the Node-Red machine.

**Comment by KK1L** [comment]
Terminal command is good hint. Will try later... halftime Eagles 49ers :)73 es God Bless de KK1L, Ron <><On Jan 29, 2023 14:36, Connor Dickey ***@***.***> wrote:
I still have a issue with what you did last time. You merge both commits which duplicated the same commits in series.  To become even with github, please force-pull (pull --force) using the terminal on the Node-Red machine.

**Comment by KK1L** [comment]
I ran the force with all up to date response. On the gui edit interface i am showing 12 commits ahead. Should I perform the pushes?73 es God Bless de KK1L, Ron <><On Jan 29, 2023 15:29, Ron Rossi ***@***.***> wrote:Terminal command is good hint. Will try later... halftime Eagles 49ers :)73 es God Bless de KK1L, Ron <><On Jan 29, 2023 14:36, Connor Dickey ***@***.***> wrote:
I still have a issue with what you did last time. You merge both commits which duplicated the same commits in series.  To become even with github, please force-pull (pull --force) using the terminal on the Node-Red machine.

**Comment by kd9lsv** [comment]
No, by doing the force pull it should have removed all outstanding commits unless you created the 12 additional commits? Your log should have the commits pictured below.

If you have trouble to start from github,  I would recommend creating a new local branch `git checkout -b <name> origin/compact_listings`. If you have extra commits, then you would need to  [cherry-pick](https://git-scm.com/docs/git-cherry-pick) the needed commits.

image

**Comment by KK1L** [comment]
Will work through this in the morning and get it straightened out. I am obviously new at git sharing73 es God Bless de KK1L, Ron <><On Jan 29, 2023 21:22, Connor Dickey ***@***.***> wrote:
No, by doing the force pull it should have removed all outstanding commits unless you created the 12 additional commits? Your log should have the commits pictured below.
If you have trouble to start from github,  I would recommend creating a new local branch git checkout -b <name> origin/compact_listings. If you have extra commits, then you would need to  cherry-pick the needed commits.

**Comment by KK1L** [comment]
Hi Connor,

I ended up creating a new local branch git checkout -b cleanup_compact_listings origin/compact_listings and now I show "Already up to date" for force pull and no pushes or pulls needed on the ui interface for sharing. So that should have gotten the pull request squared away.

I then made a few edits and will commit and sync. For SO1V and SO2R the Dashboard and RadioInfo will correctly show results. See the note below on multi-op, but y'all probably already know it is broken.

**Note:** multi-op mode is broken...not by me, but in general. In multi_op mode the station determination in Radio Info, Database, and Dashboard rely on RadioInfo.StationName which is the NetBios name of the computer sending the info. I have started to give it some though how to associate this StationName with one of the PC names someone would like to assign in node-red. You have to log the various StationNames that come across and build a database...UGH. Using RadioName is not helpful since that comes from which radio type is assigned in the Radio column of the Hardware tab of N1MM which is not unique per PC.

**Comment by kylekrieg** [comment]
Hey Ron,

Thanks for working on the dashboard and making it better.  Every little bit
helps.

My thoughts on SO1V, SO2R, Multi-Op, M/M with SO2R as there are some
challenges as we get the dashboard in more operator hands and they are
asking for enhancements or finding bugs in the original code (which is
awesome).

SO1V - single Netbios name, single radio, single VFO.  That's easy.  We've
solved that.

SO2R - single Netbios name, dual radio or single radio dual VFO.  In N1MM,
I was assuming the first radio had radio number (NR) = 1 and radio two was
NR = 2.  The dashboard would look for that single Netbios name and then
push NR = 1 to the Radio 1 position and NR = 2 to the Radio 2 position to
show both VFO's on the main dashboard screen.

Multi-OP - multiple Netbios names, multiple radios, everybody was either
running or S&P on 1 VFO only.  The dashboard would see multiple Netbios
names and Radio 1 would show up in Radio 1's info, Radio 2 and Radio 2's
into..and so on...nobody would be running multiple VFO's.  If they were,
that wasn't captured in the main dashboard screen or on the radio dashboard.

M/M with S02R - we have a couple of stations running the dashboard in a M/2
S02R configuration.  Multiple Netbios names with NR = 1 and NR = 2.  This
is mainly Flex users and big contest stations.

I'd have to really sit down and go through the flow as I haven't touched
the actual programming in a while.  Connor has completed all of the code
upgrades in the past 5 or 6 months.

73
Kyle
AA0Z

**Comment by KK1L** [comment]
I wrote some notes in my last push. I can summarize.

For SO1V and SO2R it is easy enough to map the PC Name someone chooses 
in the config page to radio 1 or radio 2. I have done that. The names 
get correctly picked up.

The multi issue is harder as you recognize. My comments from github are 
below. AFAIK the list if networked computers is not statically available 
or there apriori. Building up a list by monitoring the StationNames as 
they come in and building an indexed name list which would steer data to 
the right block on the RadioInfo tab (for example). An extra peice of 
info is available in the ContactInfo packet (NetworkedCompNr), but that 
only comes along when a contact is logged rather than every 10s as 
RadioInfo does. The Nr could be used as the index to the list of 
StationNames.

    In multi_op mode the station determination in Radio Info, Database,
    and Dashboard rely on RadioInfo.StationName which is the NetBios
    name of the computer sending the info. I have started to give it
    some though how to associate this StationName with one of the PC
    names someone would like to assign in node-red. You have to log the
    various StationNames that come across and build a database...UGH.
    Using RadioName is not helpful since that comes from which radio
    type is assigned in the Radio column of the Hardware tab of N1MM
    which is not unique per PC.

    Bottom line is it seems that assigning names to the stations from
    the config panel is problematic. Assigning a name dynamically as the
    reports come in from the various stations on the network is
    possible. There is StationName from RadioInfo and also in
    ContactInfo along with NetworkedCompNr which combined could build
    and indexed list 0 through 8 of StationNames._

73 es God Bless de KK1L
Ron <><
http://kk1l.com

**Comment by KK1L** [comment]
Have not had a chance to monkey with this in a bit...sorry. I have noticed that the CPU usage is significantly higher with this code and I have not found what of the simple changes that were made caused this issue.

**Comment by kd9lsv** [comment]
Ron,

No worries, I have been not doing much with this week. 
Note, I found the biggest issue when I had the same issue if something is looping and running constant without delay. 

I am hoping to get back on tackling more issues, but if you need additional help, let me know.

73,
Connor

**Comment by kd9lsv** [comment]
@KK1L  I found the problem with the computer latch up as one of the UI-items (tempature) would send a result upon receive data which caused an endless feedback loop.
image

**Comment by KK1L** [comment]
Dude you're kidding! Awesome. I have scratched my head on this a lot and never thought that would do it.Also shamefully I have been way to busy with work and family to monkey around much of the past two months or so.73 es God Bless de KK1L, Ron <><On Apr 8, 2023 09:54, Connor Dickey ***@***.***> wrote:
@KK1L  I found the problem with the computer latch up as one of the UI-items (tempature) would send a result upon receive data which caused an endless feedback loop.

**Comment by KK1L** [comment]
@kd9lsv  I made the change and VIOLA(!) CPU useage is back to normal...so awesome. Frankly that was one of the reasons I had not tinkered more with the flow. Silly to make changes and perpetuate a flaw. Now I can bravely move forward.

**Comment by kd9lsv** [comment]
LOL. I am just trying to get back in the swing of things and are wanting to make sure you changes get implemented. I also had that issue a long time ago, when I updated the whole thing for v1.0.0 .  

If there is anything I need to help fix to implement, please let me know.

**Comment by KK1L** [comment]
That was the outstanding issue for the compact dashboard update and having the radio names show up in Radio Info.
Getting the famous "Authentication required for repository" issue. I have to refigure out how to fix that so I can synch again.

**Comment by kd9lsv** [comment]
Make sure you are on the correct branch and pushing to the correct fork.
